mcs/class/System/Test/System.Diagnostics/PerformanceCounterTest.cs

```diff
@@ -0,0 +1,363 @@
+// This test code derives from two places:
+//
+// 1.
+// originally copied from external/corefx
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// 2. MSDN
+// https://msdn.microsoft.com/en-us/library/system.diagnostics.countercreationdata(v=vs.110).aspx
+//
+// Neither has been very significantly changed.
+//
+
+using NUnit.Framework;
+using System;
+using System.Threading;
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Specialized;
+
+namespace MonoTests.System.Diagnostics
+{
+	internal class Common
+	{
+		internal string guid;
+		internal string baseName;
+		internal string name;
+		internal string category;
+		internal CounterCreationDataCollection counterDataCollection;
+		internal CounterCreationData counter;
+		internal CounterCreationData counterBase;
+		internal PerformanceCounter counterSample;
+		internal PerformanceCounterCategory pcc;
+
+		internal void DoCommon()
+		{
+			guid = Guid.NewGuid().ToString("N");
+			name = guid + "_Counter";
+			category = name + "_Category";
+			counterDataCollection = new CounterCreationDataCollection();
+
+			// Add the counter.
+			counter = new CounterCreationData();
+			counter.CounterType = PerformanceCounterType.AverageCount64;
+			counter.CounterName = name;
+			counterDataCollection.Add(counter);
+
+			// Add the base counter.
+			counterBase = new CounterCreationData();
+			counterBase.CounterType = PerformanceCounterType.AverageBase;
+			baseName = name + "Base";
+			counterBase.CounterName = baseName;
+			counterDataCollection.Add(counterBase);
+
+			// Create the category.
+			PerformanceCounterCategory.Create(category,
+				"description",
+				PerformanceCounterCategoryType.SingleInstance, counterDataCollection);
+
+			counterSample = new PerformanceCounter(category,
+				name, false);
+
+			counterSample.RawValue = 0;
+
+			pcc = new PerformanceCounterCategory(category);
+		}
+	}
+
+	[TestFixture]
+	public class PerformanceCounterTests
+	{
+		Common a = new Common();
+
+		[SetUp]
+		public void Setup()
+		{
+			a.DoCommon();
```

Comment:
instead of the `DoCommon()` method I'd just move the logic into the constructor of the class and then just assign it here.